### PR TITLE
Issue 5650: Remove duplicate test files if present.

### LINKED ISSUE #5650: Duplicate modules cron_queue_test and system_cron_test after update
## Description of the bug

A couple of sites that have just been updated are showing Status Report error messages for:

- Cron Queue test module
- System Cron test module

There are multiple versions of module Cron Queue test in the installation. [less](https://dev.systemhorizons.co.uk/admin/reports/status#)
The competing module files are listed below. Only the first of the module versions in the list will be loaded.
**core/modules/system/tests/cron_queue_test.module**
core/modules/system/tests/cron_queue_test/cron_queue_test.module

There are multiple versions of module System Cron Test in the installation. [less](https://dev.systemhorizons.co.uk/admin/reports/status#)
The competing module files are listed below. Only the first of the module versions in the list will be loaded.
**core/modules/system/tests/system_cron_test.module**
core/modules/system/tests/system_cron_test/system_cron_test.module

I believe this was introduced through PR [#3911](https://github.com/backdrop/backdrop/pull/3911) that was aiming to address #5465 (1.21.2) but only became apparent in status report in 1.22 when #5464 was implemented.

## Steps To Reproduce

To reproduce the behavior:
1. Start with version before 1.21.2
2. Update through the UI
3. Go to status report

## Actual behavior
Update has created duplicate files

## Expected behavior
Renaming of files should have removed old files

## Additional information

Add any other information that could help, such as:
- Backdrop CMS version: 1.22
- Web server and its version: litespeed
- Operating System and its version: linux ?
- Browser(s) and their versions: N/a

### PR DESCRIPTION
**Pull request by bugfolder**
Fixes https://github.com/backdrop/backdrop-issues/issues/5650.

### CODE REVIEW

File: core/modules/system/system.install [hunk 1]
```diff
@@ -3274,6 +3274,33 @@ function system_update_1081() {
...
+    $base . 'system_cron_test.module',
+  );
+  $removed = array();
+  foreach ($filenames as $filename) {
+    if (file_exists($filename)) {
+      if (unlink($filename)) {
+        $removed[] = $filename;
+      }
+      else {
+        watchdog('system', t('Unable to remove file %filename', array('filename' => $filename)), WATCHDOG_ERROR);
```
Comment (indigoxela) [inline core/modules/system/system.install]: watchdog() in an update hook is very uncommon.

I think if something went wrong in an update hook, it's supposed to be returned directly via backdrop_set_message() and the appropriate type (warning, error).

File: core/modules/system/system.install [hunk 2]
```diff
@@ -3274,6 +3274,33 @@ function system_update_1081() {
...
+      }
+    }
+  }
+  if (!empty($removed)) {
+    return t('Removed superfluous files @files.', array('@files' => implode(', ', $removed)));
```
Comment (indigoxela) [inline core/modules/system/system.install]: It's OK to return it this way, but backdrop_set_message() is another option. Or don't return anything? Usually update hooks don't return anything if they did what they were supposed to do. (No news is good news :wink:)

File: core/modules/system/system.install (hunk 1 above)
Comment (bugfolder) [reply core/modules/system/system.install]: Thx, updated.

File: core/modules/system/system.install (hunk 2 above)
Comment (bugfolder) [reply core/modules/system/system.install]: "No news is good news" it is.

**Comment by laryn** [comment]
@bugfolder Can you rebase to fix the conflict with the update hook?

**Comment by bugfolder** [comment]
Closing in favor of rebased https://github.com/backdrop/backdrop/pull/4178.

**Pull request closed without merging**